vm/compiler/benches/coinbase_puzzle/coinbase_puzzle.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+#[macro_use]
+extern crate criterion;
+
+use console::{account::*, network::Testnet3};
+use snarkvm_compiler::{CoinbasePuzzle, EpochChallenge, EpochInfo};
+
+use criterion::Criterion;
+use rand::{self, thread_rng, CryptoRng, RngCore};
```

Comment:
when https://github.com/AleoHQ/snarkVM/pull/1013 is merged, the slow and non-deterministic `thread_rng` could be replaced with `TestRng` here

Reply:
#1013 was merged, so `TestRng` can be used now instead; it implements `CryptoRng` too, so it's compatible